conda_package_streaming/package_streaming.py

```diff
@@ -125,6 +126,9 @@ def stream_conda_component(
 
         zf = zipfile.ZipFile(fileobj or filename)
         file_id, _, _ = os.path.basename(filename).rpartition(".")
+        # this substitution compensates for web downloads from anaconda.org having
+        # the platform as a prefix
+        file_id = re.sub("^(osx|linux|win|noarch)(-.+?)?_", "", file_id)
```

Comment:
I see we don't have access to DEFAULT_SUBDIRS here, since we don't depend on conda.